### ISSUE TITLE
Dashboard - Accounts- context menu Y position

### ISSUE THREAD

**Issue by DevelopmentAnything** [moneymanagerex/moneymanagerex#6573]
MMEX version:
 - [x] 1.6.4
 - [x] 1.7.0
 - [x] 1.7.1
 - [ ] 1.7.2
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows 11
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

Please describe your issue with details.
Add screenshot or other files if needed.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/c2a7e6e0-623a-4b02-b90f-f4ec236d2307)

when right-click to account name, show pop-up menu.
However, the response is slow, so if click twice, the Delete Account menu will be selected.
(important: Pop-up menu is slow to open)

must move the top position of the shortcut menu to the position of the mouse pointer.

because most pop-up menus in Windows open based on the mouse pointer position.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/c62c3ddd-0248-4337-a7a2-ffbf1259baa8)
pop-up menu in the transaction history grid is normal.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
Dashboard Tree List
right-click to account name

Expected result:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/a4800d13-17d7-4b92-9217-bff2354c76a8)

Actual result:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/c2a7e6e0-623a-4b02-b90f-f4ec236d2307)

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/93842748-74ea-4e50-8f39-53cf27202a08)
omg 😱

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by renato-mmex** [comment]
it works very quickly for me, my computer is probably too fast <img width="28" alt="image" src="https://github.com/moneymanagerex/moneymanagerex/assets/76920138/fe53b8cc-3ed3-43db-8e2a-4fe1b186bf1b">

The cursor stops at this point and is therefore in an opening window. If you were to move this window further to the right, this problem would be solved.

**Comment by DevelopmentAnything** [comment]
Right-click not only shows the pop-up menu, but also moves to that screen (like left-click)
This is an unnecessary measure.
That's why it's slow.
and there is no reason to change the PC because of MMEX.
bcz my PC spec are i7 10850H, 64G, and NVMe SSD.

**Comment by vomikan** [comment]
This is not a bug at all, but only user actions. Windows does not have an action for double-right-click.

**Comment by DevelopmentAnything** [comment]
> This is not a bug at all, but only user actions. Windows does not have an action for double-right-click.

no no no.

Double click is what I did. pops-up slowly.
The reason is that the menu only needs to be displayed, but the screen changes too!!
The point of this article is! that the menu position is above the mouse pointer.

**Comment by DevelopmentAnything** [comment]
This is why the screen changes slowly.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/01333680-1e6f-46d8-a1af-270b9ba60bb2)

I have a list of 37 accounts

**Comment by renato-mmex** [comment]
@vomikan 
The cursor stops at this point and is therefore in an opening window. 
If you were to move this window further to the right, this problem would be solved.

<img width="306" alt="image" src="https://github.com/moneymanagerex/moneymanagerex/assets/76920138/53a7ad37-9ba4-4cc5-89e2-579104b6d0b4">
<img width="403" alt="image" src="https://github.com/moneymanagerex/moneymanagerex/assets/76920138/7e254ef9-13bf-4354-9b89-6c2a9fe898e8">

**Comment by DevelopmentAnything** [comment]
please!

내가 원하는 것은 X 위치 이동이 아닙니다!
Y 위치 입니다.
다른 팝업 메뉴는 마우스 포인터 위치에 팝업 메뉴가 나타나는데, 대시보드에서만 팝업메뉴가 마우스보다 위로 올라갑니다.
도대체 왜??!!

account tree list: popup top position = mouse Y - 123 px

Normal: popup Top = Mouse Y posotion

**Comment by renato-mmex** [comment]
translated =
_I don't want to move X position!
It's Y position.
Other popup menus appear where the mouse pointer is, but only on the dashboard does the popup menu rise above the mouse.
Why on earth?!!!_

**Comment by rmelillo76** [comment]
> > This is not a bug at all, but only user actions. Windows does not have an action for double-right-click.
> 
> no no no.
> 
> Double click is what I did. pops-up slowly. The reason is that the menu only needs to be displayed, but the screen changes too!! The point of this article is! that the menu position is above the mouse pointer. And there is no reason for the screen to change.

I think what @DevelopmentAnything  is trying to say is that the checking panel (Account View: [Account Name]) for the account that is under the mouse pointer is displayed first, and he doesn't think it should. It doesn't refresh if the account was already selected. The display of the "newly focused" account does cause a minor delay in showing the context menu. For me, the delay is trivial, but I guess that would depend on how many transactions you have and the filter you had pre-defined on the panel.

There are other places where this happens too, for example right click on transaction reports is the only way to edit, rename or delete but you have to wait for it to render as-is first:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/95182117/e8449eaf-d636-461c-a268-c29131abf507)

**Comment by DevelopmentAnything** [comment]
I've registered over 30 accounts, over 7600 transactions. popup menus open slowly even in accounts that are already open.

But!
popup menus in the transactions list open very quickly.
menu also appears with row not selected

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/c355842a-63eb-4f5c-b1cb-8cb7ec193ee4)

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/a78aa225-edfd-4dde-b63f-fe35f05c1ab4)

Look at the mouse pointer and menu position in the picture above.

Please review the performance problem of the menu appearing late later.

**Comment by ovari** [comment]
@DevelopmentAnything can you please test the development version. If this issue is resolved, can you please close this issue? If this issue has not been resolved, can you please add a comment. Thank you

**Comment by DevelopmentAnything** [comment]
@ovari
context menu Y location is normal. OK.
But I keep seeing strange movements.
As soon as right-click(account list), the current screen is updated once, changed to the screen of the account, and renewed twice again.
the next context menu appears.
Grid flashes as it renews, and the processing speed is quite slow.

There seems to be a problem with the shortcut menu event handler.

**Comment by DevelopmentAnything** [comment]
It would be better to deal with this as a separate issue.
